lms/static/js/bookmarks/views/bookmark_button.js

```diff
@@ -1,6 +1,6 @@
 ;(function (define, undefined) {
     'use strict';
-    define(['gettext', 'jquery', 'underscore', 'backbone', 'js/views/message_banner'],
+    define(['gettext', 'jquery', 'underscore', 'backbone', 'lmsCoursewareMessageBanner'],
```

Comment:
This is going to be problematic for RequireJS Optimizer which won't know where this file is when it bundles up the factory. The configuration setting you added will only apply at runtime, so won't help Optimizer. My recommendation of leaving the paths as is will help in this situation too, as Optimizer will be able to find them as normal.